client-sdk/go/config/network.go

```diff
@@ -137,7 +138,17 @@ func (n *Network) Validate() error {
 	return nil
 }
 
-// IsLocalRPC checks whether the RPC endpoint points to a local UNIX socket.
+// IsLocalRPC checks whether the RPC endpoint points to a local UNIX socket or loopback address.
 func (n *Network) IsLocalRPC() bool {
-	return strings.HasPrefix(n.RPC, "unix:")
+	if strings.HasPrefix(n.RPC, "unix:") {
+		return true
+	}
+
+	// RPC addresses are without scheme.
+	host, _, err := net.SplitHostPort(n.RPC)
+	if err != nil {
+		return false
+	}
+
+	return net.ParseIP(host).IsLoopback() || host == "localhost"
```

Comment:
You may also have virutal hosts or non-standard names for 127.0.0.1 right?

```suggestion
	ip, err := net.LookupIP(host)
	if err != nil || ip.length==0 {
		return false
	}
	return ip.IsLoopback()
```